File: android/src/com/dontflymoney/auth/Authentication.java
package com.dontflymoney.auth;

import android.content.Context;

public class Authentication
{
	Context context;
	
	public Authentication(Context context)
	{
		this.context = context;
	}
	
	public void Set(String ticket)
	{
		File ticketFile = new File(context, File.Ticket);
		
		ticketFile.WriteToFile(ticket);
	}
	
	public String Get()
	{
		File ticketFile = new File(context, File.Ticket);
		
		return ticketFile.ReadFromFile();
	}

	public boolean IsLoggedIn()
	{
		String ticket = Get();

		return ticket != null && ticket != "";
	}

	public void Clear()
	{
		Set(null);
	}
	

}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/auth/Authentication.kt
package com.darakeon.dfm.auth

import android.content.Context

class Authentication(private val context: Context) {
	private val aks = Aks(context)
	private val ticketKey = "Ticket"
	private val admKey = "Ticket"

	var ticket: String
		get() {
			val encryptedTicket = context.getValue(ticketKey)
			return aks.decrypt(encryptedTicket)
		}
		set(value) {
			val encryptedTicket = aks.encrypt(value)
			context.setValue(ticketKey, encryptedTicket)
		}

	var isAdm: Boolean
		get() {
			val encryptedTicket = context.getValue(admKey)
			return aks.decrypt(encryptedTicket) == "true"
		}
		set(value) {
			val encryptedTicket = aks.encrypt(value.toString())
			context.setValue(admKey, encryptedTicket)
		}

	fun clear() {
		ticket = ""
		isAdm = false
	}

	val isLoggedIn
		get() = ticket != ""
}
